### PR TITLE
Feature/ab#1023 python java support

### PR DESCRIPTION
**Pull request by Anthony-Mckale**
- updated cyclonedx plugin
- added section of asset caching to README.md
- updated cyclonedx to support poetry/piplock and requirements.txt
- extended piprot/cyclonedx/safety to all support multi project environments
- updated linux base image to python 3.9 and buster -> bullseye

testable with https://github.com/RiverSafeUK/eze-goats/tree/main/goat-python/python-autorun

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -61,6 +61,9 @@ cli: cli-build cli-install
 docker: cli-build cli-install
 	docker build --tag eze-cli .
 
+# clean up docker images, remove all unused images
+docker-clean: docker system prune -f
```
Comment (mgonzales-rs) [inline Makefile:65]: one qs for me to learn, so I used to do `docker rm -f $(docker ps -a -q)` to delete my container and the docker rmi to delete my images, will this command delete all docker images, then?

File: README.md [hunk 2]
```diff
@@ -297,6 +297,21 @@ f0bef6e0bba7   optimistic_burnell   0.01%     104.8MiB / 12.33GiB   0.83%     22
 $ docker rm --force f0bef6e0bba7
 ```
 
+## Asset Caching
+Many tools such as maven and npm download depedencies from the internet
```
Comment (mgonzales-rs) [inline README.md:301]: lil typo: dependencies*

**Review by mgonzales-rs — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (Anthony-Mckale) [reply README.md:301]: added to AB#1027 PR

File: Makefile (hunk 1 above)
Comment (Anthony-Mckale) [reply Makefile:65]: i think rm stops images, prune deleted the unused ones, it saved gigs of hard disk space after running it for the first time in months

**Pull request merged**